internal/verification/verification.go

```diff
@@ -120,56 +125,67 @@ func (v Validator) VerifyDataIntegrityCredential(ctx context.Context, credential
 	return v.staticValidationChecks(ctx, credential)
 }
 
-func getKeyFromProof(proof crypto.Proof, key string) (any, error) {
-	proofBytes, err := json.Marshal(proof)
+// VerifyJWTPresentation first parses and checks the signature on the given JWT presentation. Next, it runs
+// a set of static verification checks on the presentation's credentials as per the service's configuration.
+func (v Verifier) VerifyJWTPresentation(ctx context.Context, token keyaccess.JWT) error {
+	headers, jwt, vp, err := integrity.ParseVerifiablePresentationFromJWT(token.String())
 	if err != nil {
-		return nil, err
-	}
-	var proofMap map[string]any
-	if err = json.Unmarshal(proofBytes, &proofMap); err != nil {
-		return nil, err
+		return errors.Wrap(err, "parsing JWT presentation")
 	}
-	return proofMap[key], nil
-}
 
-func (v Validator) VerifyJWT(ctx context.Context, did string, token keyaccess.JWT) error {
-	// parse headers
-	headers, err := keyaccess.GetJWTHeaders([]byte(token))
-	if err != nil {
-		return sdkutil.LoggingErrorMsg(err, "could not parse JWT headers")
+	// get key to verify the presentation with
+	issuerKID := headers.KeyID()
+	if issuerKID == "" {
+		return errors.Errorf("missing kid in header of presentation<%s>", jwt.JwtID())
 	}
-	jwtKID, ok := headers.Get(jws.KeyIDKey)
-	if !ok {
-		return sdkutil.LoggingNewError("JWT does not contain a kid")
+	issuerDID, err := v.didResolver.Resolve(ctx, jwt.Issuer())
+	if err != nil {
+		return errors.Wrapf(err, "error getting issuer DID<%s> to verify presentation<%s>", jwt.Issuer(), jwt.JwtID())
 	}
-	kid, ok := jwtKID.(string)
-	if !ok {
-		return sdkutil.LoggingNewError("JWT kid is not a string")
+	issuerKey, err := did.GetKeyFromVerificationMethod(issuerDID.Document, issuerKID)
```

Comment:
`issuerPublicKey` for clarity of future readers.

Reply:
done